### ISSUE TITLE
Crash on iOS 15 

### ISSUE THREAD

**Issue by alrocha** [Eppo-exp/eppo-ios-sdk#46]
Hi eppo team, 

I am opening this issue due to a crash encountered from version 3.1.0 in your repository compiling the app with iOS 15. With 3.0.1 everything works fine. 

```
dyld[3124]: Symbol not found: (_$sSo17OS_dispatch_queueC8DispatchE12asyncAndWait5flags7executexAC0D13WorkItemFlagsV_xyKXEtKlF)
```
We are getting a crash on start of the app, let me know if you need more details.

**Comment by leoromanovsky** [comment]
@alrocha Thanks for the note. In release `3.2.0` we made an improvement to the initialization (https://github.com/Eppo-exp/eppo-ios-sdk/commit/f65ace1f6f40fcdf3a098dad1de55229f002f79d).

Could you please try the latest tag `3.3.0` and see if that addresses this issue.

**Comment by alrocha** [comment]
I have tested with `3.1.0`, `3.2.0` and `3.3.0` and the crash happens in all of them with the error 

```
dyld[3870]: Symbol not found: (_$sSo17OS_dispatch_queueC8DispatchE12asyncAndWait5flags7executexAC0D13WorkItemFlagsV_xyKXEtKlF)
  Referenced from: '/private/var/containers/Bundle/Application/DD1510CD-F85B-4F2A-B294-1B6DB0064196/AppName-Debug.app/AppName-Debug.debug.dylib'
  Expected in: '/usr/lib/swift/libswiftDispatch.dylib'
```

**Comment by leoromanovsky** [comment]
@alrocha We are implementing a fix for this issue (PR linked above).

One thing I want to clarify with you is your report that the issue was seen in `3.1.0`.

👉 My belief is that the issue was introduced in (https://github.com/Eppo-exp/eppo-ios-sdk/commit/f65ace1f6f40fcdf3a098dad1de55229f002f79d) which was only published in `3.2.0` and `3.3.0`, before than we did not use the `asyncAndWait` method of `DispatchQueue`.

I plan to publish the fix to `3.2.1` and `3.3.1` but want to confirm from your notes about `3.1.0` if this is correct. Thank you.

**Comment by leoromanovsky** [comment]
Resolved and backported to the `3.2` and `3.3` releases with:

* https://github.com/Eppo-exp/eppo-ios-sdk/releases/tag/3.2.1
* https://github.com/Eppo-exp/eppo-ios-sdk/releases/tag/3.3.1

**Comment by alrocha** [comment]
Thanks a lot @leoromanovsky the latest version works! 🚀 
Apologies for my missing replies since I have been a few weeks off.
